File: go/0027E-移除元素.go
/*
 * @Author: SukiEva
 * @Date: 2021-10-09 16:35:34
 * @Description: https://github.com/SukiEva
 * わたし、気になります！
 */

package main

func removeElement(nums []int, val int) int {
	k := 0
	for i := 0; i < len(nums); i++ {
		if nums[i] != val {
			nums[k] = nums[i]
			k++
		}
	}
	return k
}

File: cpp/0027E-移除元素.cpp
/*
 * @Author: SukiEva
 * @Date: 2021-12-13 09:09:54
 * @Description: https://github.com/SukiEva
 * わたし、気になります！
 */
#include <bits/stdc++.h>

using namespace std;

class Solution {
   public:
    int removeElement(vector<int>& nums, int val) {
        int k = 0;
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] != val) {
                nums[k++] = nums[i];
            }
        }
        return k;
    }
};